tests/aequilibrae/project/test_field_editor.py

```diff
@@ -1,137 +1,101 @@
-            del curr
-
-    def test_check_completeness(self):
-        for table in self.my_tables:
-            curr = self.proj.conn.cursor()
-
-            # We add a bogus record to the attribute list
-            val = self.randomword(30).lower()
-            qry = 'INSERT INTO "attributes_documentation" VALUES (?,?," ");'
-            curr.execute(qry, (table, val))
-            self.proj.conn.commit()
-
-            curr.execute(f'Select name_table from "attributes_documentation" where attribute="{val}"')
-            self.assertEqual(curr.fetchone()[0], table, "Failed to insert bogus value")
-
-            # Then we add a new field to the table
-            val2 = self.randomword(10)
-            curr.execute(f'Alter table "{table}" add column "{val2}" NUMERIC;')
-            curr.execute(f"pragma table_info({table})")
-            fields = [x[1] for x in curr.fetchall() if x[1] == val2]
-            self.assertEqual([val2], fields, "failed to add a new field")
-
-            table = FieldEditor(self.proj, table)
-            self.proj.conn.commit()
-            curr = self.proj.conn.cursor()
-            curr.execute(f'Select count(*) from "attributes_documentation" where attribute="{val}"')
-            self.assertEqual(curr.fetchone()[0], 0, f"clean the table on loading failed {val}")
-
-            curr.execute(f'Select count(*) from "attributes_documentation" where attribute="{val2}"')
-            self.assertEqual(curr.fetchone()[0], 1, "clean the table on loading failed")
-            print(table._original_values[val2])
-
-            self.proj.conn.commit()
-            del curr
-
-    def test_save(self):
-        for tab in ["modes", "links", "nodes", "link_types"]:
-            table = FieldEditor(self.proj, tab)
-            random_val = self.randomword(30)
-            if "alpha" in table._original_values.keys():
-                table.alpha = random_val
-                table.save()
-                table2 = FieldEditor(self.proj, tab)
-
-                self.assertEqual(table2.alpha, random_val, "Did not save values properly")
-
-            if "link_id" in table._original_values.keys():
-                table.link_id = random_val
-                table.save()
-                table2 = FieldEditor(self.proj, tab)
-
-                self.assertEqual(table2.link_id, random_val, "Did not save values properly")
-
-            if "node_id" in table._original_values.keys():
-                table.node_id = random_val
-                table.save()
-                table2 = FieldEditor(self.proj, tab)
-
-                self.assertEqual(table2.node_id, random_val, "Did not save values properly")
-
-            self.proj.conn.commit()
+from pathlib import Path
+
+import pytest
+from aequilibrae.project.field_editor import FieldEditor
+
+
+class TestFieldEditor:
+    @pytest.fixture(scope="session")
+    def project_session(self, create_empty_project_session):
+        return create_empty_project_session()
```

Comment:
Would it make sense to now move this next to the `project` fixture in the conftest.py?

Reply:
I'm a little bit hesitant to do this. Having a session scoped project inside a class is one thing, having a global sessions scoped project is anothe. Currently, if a test behaves poorly and doesn't clean up after itself, it can only affects the other tests inside the class. In case of a global fixture that's used throughout the whole test suite there's an increased risk of tests affecting other tests in unpredictable ways.

Also, in the `TestFieldEditor` class we know the only part of the Project that is mutated is the sqlite database, so we also only need to restore that. For the general case, we don't know that, so we also cannot make the assumption that restoring the database is sufficient. So we'd need to restore the whole project, which is a significantly more expensive operation (and is effectively what we do with `create_project_session` already)

Reply:
Hmmm - my understanding of scope is that if it's defined with session scope - it won't matter where it's defined - there will only be one of them for the entire test suite. Or are you relying here on the fact that each of the fixtures is namespaced to the class that is invoking it? I think that makes sense - although it could therefore be tagged as "module" level scope.

Reply:
If what I said lines up with your understanding - i'm happy to resolve this conversation and approve.